SourceWriters/BitmapEffectWriter.cs

```diff
@@ -0,0 +1,188 @@
+﻿using PaintDotNet.Effects.Gpu;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Security;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace PaintDotNet.Effects
+{
+    internal static class BitmapEffectWriter
+    {
+        internal const string UsingStatements = ""
+            + "using System;\r\n"
+            + "using System.IO;\r\n"
+            + "using System.Linq;\r\n"
+            + "using System.Diagnostics;\r\n"
+            + "using System.Drawing;\r\n"
+            + "using System.Threading;\r\n"
+            + "using System.Reflection;\r\n"
+            + "using System.Drawing.Text;\r\n"
+            + "using System.Windows.Forms;\r\n"
+            + "using System.IO.Compression;\r\n"
+            + "using System.Drawing.Drawing2D;\r\n"
+            + "using System.Collections.Generic;\r\n"
+            + "using System.Text;\r\n"
+            + "using System.Text.RegularExpressions;\r\n"
+            + "using System.Runtime.InteropServices;\r\n"
+            + "using System.Runtime.Versioning;\r\n"
+            + "using Registry = Microsoft.Win32.Registry;\r\n"
+            + "using RegistryKey = Microsoft.Win32.RegistryKey;\r\n"
+            + "using PaintDotNet;\r\n"
+            + "using PaintDotNet.AppModel;\r\n"
+            + "using PaintDotNet.Direct2D1;\r\n"
+            + "using PaintDotNet.Effects;\r\n"
+            + "using PaintDotNet.Clipboard;\r\n"
+            + "using PaintDotNet.Imaging;\r\n"
+            + "using PaintDotNet.IndirectUI;\r\n"
+            + "using PaintDotNet.Collections;\r\n"
+            + "using PaintDotNet.PropertySystem;\r\n"
+            + "using PaintDotNet.Rendering;\r\n"
+            + "using ColorWheelControl = PaintDotNet.ColorBgra;\r\n"
```

Comment:
I definitely recommend to use `ColorBgra32` instead of `ColorBgra`. I might need to make sure that some of the boxed data transfers work (e.g. setting the value of an IndirectUI `Property`).